lib/listen/record.rb

```diff
@@ -46,16 +46,18 @@ def file_data(rel_path)
     end
 
     def dir_entries(rel_path)
-      subtree = if ['', '.'].include? rel_path.to_s
+      rel_path_s = rel_path.to_s
+      subtree = if ['', '.'].include?(rel_path_s)
```

Comment:
I would suggest moving the assignment into the expression for readability sake.

Reply:
Which assignment? `rel_path_s =`?

Reply:
I would personally prefer:

```
if ...
  subtree = ... 
else
  subtree = ...
end
```

I think it's easier to read and less likely to cause bugs in the future.

Reply:
Hmm. To me, that would be a surprising step away from DRY code. I've found the common assignment can be less bug-prone, for example if an `elsif` is added. And it often identifies a good refactor point where the entire `if`/`else` cascade can move into a separate method. I'd rather leave this for now.